src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1907,62 +1826,75 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
         Set<String> studentsEmailInTeam = new HashSet<String>();
         if (role == UserRole.STUDENT) {
             student = studentsLogic.getStudentForEmail(courseId, userEmail);
-            List<StudentAttributes> studentsInTeam = studentsLogic
-                    .getStudentsForTeam(student.team, courseId);
+            List<StudentAttributes> studentsInTeam = studentsLogic.getStudentsForTeam(student.team, courseId);
             for (StudentAttributes teammates : studentsInTeam) {
                 studentsEmailInTeam.add(teammates.email);
             }
         }
-        
+
         InstructorAttributes instructor = null;
         if (role == UserRole.INSTRUCTOR) {
             instructor = instructorsLogic.getInstructorForEmail(courseId, userEmail);
         }
         
         Map<String, FeedbackResponseAttributes> relevantResponse = new HashMap<String, FeedbackResponseAttributes>();
         for (FeedbackResponseAttributes response : allResponses) {
-            FeedbackQuestionAttributes relatedQuestion = allQuestionsMap
-                    .get(response.feedbackQuestionId);
+            FeedbackQuestionAttributes relatedQuestion = allQuestionsMap.get(response.feedbackQuestionId);
             if (relatedQuestion != null) {
-                boolean isVisibleResponse = isResponseVisibleForUser(userEmail,
-                        role, student, studentsEmailInTeam, response,
-                        relatedQuestion, instructor);
+                boolean isVisibleResponse = isResponseVisibleForUser(userEmail, role, student, studentsEmailInTeam,
+                                                                     response, relatedQuestion, instructor);
                 if (isVisibleResponse) {
                     responses.add(response);
                     relevantResponse.put(response.getId(), response);
-                    relevantQuestions.put(relatedQuestion.getId(),
-                            relatedQuestion);
-                    addEmailNamePairsToTable(emailNameTable, response,
-                            relatedQuestion, roster);
-                    addEmailLastNamePairsToTable(emailLastNameTable, response,
-                            relatedQuestion, roster);
-                    addEmailTeamNamePairsToTable(emailTeamNameTable, response,
-                            relatedQuestion, roster);
-                    addVisibilityToTable(visibilityTable, relatedQuestion,
-                            response, userEmail, role, roster);
+                    relevantQuestions.put(relatedQuestion.getId(), relatedQuestion);
```

Comment:
the adding of emails and visibility to tables can also be refactored since `getFeedbackSessionResultsForQuestionId` handles a visible response as well